.pre-commit-config.yaml

```diff
@@ -0,0 +1,84 @@
+repos:
+  # General
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.2.0
+    hooks:
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
+      - id: mixed-line-ending
+        args: ["--fix=lf"]
+      - id: check-merge-conflict
+      - id: check-toml
+      - id: check-yaml
+      - id: check-ast
+      - id: check-added-large-files
+  # Fish
+  - repo: local
+    hooks:
+      - id: fish_syntax
+        name: Fish syntax check
+        description: Check Fish shell syntax
+        entry: fish --no-execute
+        language: system
+        files: "(fish)"
+      - id: fish_indent
+        name: Fish indent check
+        description: Check Fish shell indentation
+        entry: fish_indent --write
+        language: system
+        files: "(fish)"
+  # Misc
+  - repo: https://github.com/Lucas-C/pre-commit-hooks
+    rev: v1.1.10
+    hooks:
+      - id: insert-license
+        args: [--license-filepath, LICENSE_BOILERPLATE, "--detect-license-in-X-top-lines=20"]
+        exclude: '^(.github|.git)'
```

Comment:
what about markdown files?

Reply:
Doesn't seem to wreck anything yet. I'll add it when it breaks it.